cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -1189,7 +1192,12 @@ func newServer(conf rootConfig) (*Server, error) {
 			}
 		}
 
-		sg = newSourcegraphClient(rootURL, conf.hostname, batchSize)
+		port, err := toPort(conf.listen)
```

Comment:
if we are listening on a port, we can get the well formatted address from the net.Listener so you don't need to have a custom toPort